source/FreeRTOS_Routing.c

```diff
@@ -97,35 +97,38 @@ struct xIPv6_Couple
-        pxEndPoint->ipv4_settings.ulBroadcastAddress = ulIPAddress | ~( pxEndPoint->ipv4_settings.ulNetMask );
-
-        /* Copy the current values to the default values. */
-        ( void ) memcpy( &( pxEndPoint->ipv4_defaults ), &( pxEndPoint->ipv4_settings ), sizeof( pxEndPoint->ipv4_defaults ) );
-
-        /* The default IP-address will be used in case DHCP is not used, or also if DHCP has failed, or
-         * when the user chooses to use the default IP-address. */
-        pxEndPoint->ipv4_defaults.ulIPAddress = ulIPAddress;
-
-        #if ( ipconfigUSE_DHCP != 0 )
-            if( pxEndPoint->bits.bWantDHCP == 0U )
-        #endif
+        if( ( pxNetworkInterface == NULL ) || ( pxEndPoint == NULL ) )
         {
-            pxEndPoint->ipv4_settings.ulIPAddress = ulIPAddress;
+            /* Invalid input. */
+            FreeRTOS_printf( ( "FreeRTOS_FillEndPoint: Invalid input, netif=%p, endpoint=%p\n",
+                               pxNetworkInterface,
+                               pxEndPoint ) );
         }
+        else
+        {
+            /* Fill in and add an end-point to a network interface.
+             * The user must make sure that the object pointed to by 'pxEndPoint'
+             * will remain to exist. */
+            ( void ) memset( pxEndPoint, 0, sizeof( *pxEndPoint ) );
+
+            ulIPAddress = FreeRTOS_inet_addr_quick( ucIPAddress[ 0 ], ucIPAddress[ 1 ], ucIPAddress[ 2 ], ucIPAddress[ 3 ] );
+            pxEndPoint->ipv4_settings.ulNetMask = FreeRTOS_inet_addr_quick( ucNetMask[ 0 ], ucNetMask[ 1 ], ucNetMask[ 2 ], ucNetMask[ 3 ] );
+            pxEndPoint->ipv4_settings.ulGatewayAddress = FreeRTOS_inet_addr_quick( ucGatewayAddress[ 0 ], ucGatewayAddress[ 1 ], ucGatewayAddress[ 2 ], ucGatewayAddress[ 3 ] );
+            pxEndPoint->ipv4_settings.ulDNSServerAddresses[ 0 ] = FreeRTOS_inet_addr_quick( ucDNSServerAddress[ 0 ], ucDNSServerAddress[ 1 ], ucDNSServerAddress[ 2 ], ucDNSServerAddress[ 3 ] );
+            pxEndPoint->ipv4_settings.ulBroadcastAddress = ulIPAddress | ~( pxEndPoint->ipv4_settings.ulNetMask );
+
+            /* Copy the current values to the default values. */
+            ( void ) memcpy( &( pxEndPoint->ipv4_defaults ), &( pxEndPoint->ipv4_settings ), sizeof( pxEndPoint->ipv4_defaults ) );
 
-        /* The field 'ipv4_settings.ulIPAddress' will be set later on. */
+            /* The default IP-address will be used in case DHCP is not used, or also if DHCP has failed, or
+             * when the user chooses to use the default IP-address. */
+            pxEndPoint->ipv4_defaults.ulIPAddress = ulIPAddress;
 
-        ( void ) memcpy( pxEndPoint->xMACAddress.ucBytes, ucMACAddress, sizeof( pxEndPoint->xMACAddress ) );
-        ( void ) FreeRTOS_AddEndPoint( pxNetworkInterface, pxEndPoint );
+            /* The field 'ipv4_settings.ulIPAddress' will be set later on. */
```

Comment:
[Similar case](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/845#discussion_r1189617806) for IPv4 endpoints as well

Reply:
Same as comment [below](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/845/files/c80762bb530940fda9c9fc58ed1c0647059ae601#r1193305558).

Reply:
Here it is not so important because an IPv4 endpoint doesn't normally need another IPv4 endpoint.